### PR TITLE
Tokenization

### PR DESCRIPTION
**Pull request by ba1ash**
* Introduce tokenization transaction type
    Responsible for collecting 3-D Secure Data and credit card tokenization
*  Simplify and refactor cod
     Remove duplicated calls for #make_response

### CODE REVIEW

File: spec/be_gateway/client_spec.rb [hunk 1]
```diff
@@ -200,26 +200,82 @@
...
+
+      context 'child' do
         %w(capture void refund).each do |tr_type|
           describe "##{tr_type}" do
-            let(:request_params) {
+            let(:request_params) do
               {}.tap do |hsh|
                 hsh['parent_uid'] = '4107-310b0da80b',
-                hsh['amount'] = 100
+                                    hsh['amount'] = 100
```
Comment (shiroginne) [inline spec/be_gateway/client_spec.rb]: hm?

Comment (ba1ash) [inline spec/be_gateway/client_spec.rb]: We have a strange comma on the line above..

**Comment by ba1ash** [comment]
🆙

File: spec/be_gateway/client_spec.rb [hunk 2]
```diff
@@ -23,9 +23,9 @@
       let(:request_params) do
         {
           'request' => {
-            'number' => 4200000000000000,
-            'holder' => "John Smith",
-            'exp_month' => 05,
+            'number' => '4200000000000000',
+            'holder' => 'John Smith',
+            'exp_month' => 0o5,
```
Comment (AllanQ) [inline spec/be_gateway/client_spec.rb]: Is it mistake?

Comment (shiroginne) [inline spec/be_gateway/client_spec.rb]: ➕

File: lib/be_gateway/client.rb [hunk 3]
```diff
@@ -1,54 +1,49 @@
...
     def v2_update_card_by_token(token, params)
-      response = put "/v2/credit_cards/#{token}", { request: params }
-      make_response(response)
+      put("/v2/credit_cards/#{token}", request: params)
     end
 
     private
 
-    def post_url(tr_type)
+    def transaction_url(tr_type)
```
Comment (seongreen) [inline lib/be_gateway/client.rb]: Rename to operation_url or action_url. Transaction it's fin term.

File: lib/be_gateway/connection.rb [hunk 4]
```diff
@@ -38,11 +38,11 @@ def get(path)
     end
 
     def send_request(method, path, params = nil)
-      begin
-        connection.public_send(method, path, params)
-      rescue Faraday::Error::ClientError
-        OpenStruct.new(
-          {
+      handle_response(
```
Comment (shiroginne) [inline lib/be_gateway/connection.rb]: I little bit concern about this code 'style', some suggestions:
```
response = begin
...
handle_response(response)
```
or
```
handle_response do
... do your code
end
```
but then you have to change `handle_response` signature to accept block and then 
```
response = yeild
(200..299).include?(response.status) ? Response.new(response.body) : ErrorResponse.new(response.body)
```

Comment (ba1ash) [inline lib/be_gateway/connection.rb]: agreed, will be changed

File: lib/be_gateway/client.rb (hunk 3 above)
Comment (ba1ash) [inline lib/be_gateway/client.rb]: agreed

File: spec/be_gateway/client_spec.rb (hunk 2 above)
Comment (ba1ash) [inline spec/be_gateway/client_spec.rb]: `05` is an octal number. `0o5` is the same. where is mistake? 
I suppose the string should be here. Will be fixed.

**Comment by ba1ash** [comment]
🆙

File: lib/be_gateway/connection.rb [hunk 5]
```diff
@@ -14,69 +14,47 @@ def initialize(params)
...
-          connection.public_send(method, path, params)
-        rescue Faraday::Error::ClientError
-          OpenStruct.new(
-            status: 500,
-            body: {
-              'response' => {
-                'message' => 'Gateway is temporarily unavailable',
-                'errors' => {
-                  'gateway' => 'is temporarily unavailable'
+      r = begin
```
Comment (shiroginne) [inline lib/be_gateway/connection.rb]: do not use 'r' or 'x' or other short variable names, it's against the guidelines and unreadable

File: lib/be_gateway/connection.rb [hunk 6]
```diff
@@ -14,69 +14,47 @@ def initialize(params)
...
-        end
-      )
+            )
+          end
+      (200..299).cover?(r.status) ? Response.new(r.body) : ErrorResponse.new(r.body)
     end
 
     def connection
-      @connection ||= Faraday.new(url, options || {}) do |conn|
+      @connection ||= Faraday.new(url, opts || {}) do |conn|
```
Comment (shiroginne) [inline lib/be_gateway/connection.rb]: cant see any reason for renaming 'options' to 'opts'

File: lib/be_gateway/connection.rb (hunk 5 above)
Comment (ba1ash) [inline lib/be_gateway/connection.rb]: r -> response. we are in the send_request method. so, we operate with response inside it. obvious 😄

File: lib/be_gateway/connection.rb (hunk 6 above)
Comment (ba1ash) [inline lib/be_gateway/connection.rb]: method `#logger`, its one-line implementation is bigger than 80 symbols with `options` 😸

File: lib/be_gateway/connection.rb (hunk 5 above)
Comment (shiroginne) [inline lib/be_gateway/connection.rb]: bad practice ;)

**Pull request merged**